sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java

```diff
@@ -252,6 +266,29 @@ public DefaultAzureCredential build() {
                 "Only one of managedIdentityClientId and managedIdentityResourceId can be specified."));
         }
 
+        // Check required environment variables
+        if (requiredEnvVars != null && requiredEnvVars.length > 0) {
+            Configuration configuration = identityClientOptions.getConfiguration() == null
+                ? Configuration.getGlobalConfiguration().clone()
+                : identityClientOptions.getConfiguration();
+
+            List<String> missingVars = new ArrayList<>();
+            for (String envVar : requiredEnvVars) {
+                if (CoreUtils.isNullOrEmpty(configuration.get(envVar))) {
+                    missingVars.add(envVar);
+                }
+            }
+
+            if (!missingVars.isEmpty()) {
+                throw LOGGER.logExceptionAsError(new IllegalStateException(
+                    "Required environment variables are missing: " + String.join(", ", missingVars)
+                        + " Ensure these environment variables are set before creating the DefaultAzureCredential."
+                        + " See https://aka.ms/azure-identity-java-default-azure-credential-troubleshoot for more information."
```

Comment:
Use this aka.ms link instead. It follows the same pattern as the TSG links for other credentials.
```suggestion
                        + " See https://aka.ms/azsdk/java/identity/defaultazurecredential/troubleshoot for more information."
```